SpiceSharp/Components/Waveforms/Pwl.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.Collections.Generic;
+using SpiceSharp.IntegrationMethods;
+using SpiceSharp.Simulations;
+
+namespace SpiceSharp.Components.Waveforms
+{
+    public class Pwl : Waveform
+    {
+        public Pwl(double[] times, double[] voltages, double initialVoltage = 0.0)
+        {
+            if (times == null)
+            {
+                throw new NullReferenceException(nameof(times));
+            }
+
+            if (voltages == null)
+            {
+                throw new NullReferenceException(nameof(voltages));
+            }
+
+            if (times.Length != voltages.Length)
+            {
+                throw new InvalidOperationException("PWL - times array has different length than voltages array");
+            }
+
+            Points = CreatePoints(times, voltages, initialVoltage);
+            LineParameters = CreateLineParameters(Points);
+        }
+
+        public List<Point> Points { get; }
+
+        protected LineDefinition[] LineParameters { get; }
+
+        public override void Accept(TimeSimulation simulation)
+        {
+            if (simulation == null)
+            {
+                throw new ArgumentNullException(nameof(simulation));
+            }
+
+            double time = simulation.Method.Time;
+            if (time.Equals(0.0))
+                Value = Points[0].Y;
+
+            if (simulation.Method is IBreakpoints method)
+            {
+                var breaks = method.Breakpoints;
+                if (!method.Break)
+                    return;
+
+                for (int i = 0; i < Points.Count; i++)
+                {
+                    breaks.SetBreakpoint(Points[i].X);
+                }
+                   
+            }
+        }
+
+        public override void Probe(TimeSimulation simulation)
+        {
+            var time = simulation.Method.Time;
+
+            Value = GetLineValue(Points, LineParameters, time);
+        }
+
+        public override void Setup()
+        {
+        }
+
+        protected List<Point> CreatePoints(double[] times, double[] voltages, double initialVoltage)
+        {
+            var result = new List<Point>();
+
+            for (var i = 0; i < times.Length; i++)
+            {
+                result.Add(new Point(times[i], voltages[i]));
+            }
+
+            if (times[0] != 0.0)
+            {
+                result.Insert(0, new Point(0.0, initialVoltage));
+            }
+
+            return result;
+        }
+
+        protected static double GetLineValue(List<Point> points, LineDefinition[] lines, double x)
+        {
+            int index = 0;
```

Comment:
Can be optimized: When a timepoint is accepted you can assume that there will not be an `x` smaller than than the accepted timepoint. So you can find an index > 0 to start searching from.